test/timetable.test.js

```diff
@@ -0,0 +1,208 @@
+  const timeline = {
+    start: 27000,
+    end: 77400,
+    duration: 50400,
+    hourDuration: 3600,
+    hours: 14,
+    firstHour: 8,
+    offset: 0.5,
+  }
+
+  const expected = [
+    {
+      id: 'hlbl-0',
+      label: 8,
+      position: 0.03571428571428571,
+      length: 0.07142857142857142,
+    },
+    {
+      id: 'hlbl-1',
+      label: 9,
+      position: 0.10714285714285714,
+      length: 0.07142857142857142,
+    },
+    {
+      id: 'hlbl-12',
+      label: 20,
+      position: 0.8928571428571429,
+      length: 0.07142857142857142,
+    },
+    {
+      id: 'hlbl-13',
+      label: 21,
+      position: 0.9642857142857143,
+      length: 0.07142857142857142,
+    },
+  ]
+
+  const result = tt.calculateHourLabels(timeline)
+  const rlen = result.length
+
+  // Test n, n+1, m-1, m (n=first, m=last)
+  const resultSelection = [result[0], result[1], result[rlen - 2], result[rlen - 1]]
+
+  t.deepEqual(resultSelection, expected, 'calculate hour labels for timeline 7:30 - 21:15')
+  t.end()
+})
+
+test('mapPropertiesToClass()', t => {
+  const properties = {
+    isA: true,
+    isB: false,
+    isProperty: true,
+    isLoremIpsum: false,
+    isDolorSitAmet: true,
+  }
+
+  const expected = 'class class--a class--property class--dolorsitamet'
+
+  t.equal(tt.mapPropertiesToClass(properties, 'class'), expected, 'returns properties transformed to classname')
+  t.end()
+})
+
+test('groupEventsByDays()', t => {
+  const events = [
+    {startsAt: '2015-12-03 12:00:00'},
+    {startsAt: '2015-12-03 14:00:00'},
+    {startsAt: '2015-12-04 12:00:00'},
+    {startsAt: '2015-12-05 12:00:00'},
+  ]
+
+  const expected = {
+    3: [
+      {startsAt: '2015-12-03 12:00:00'},
+      {startsAt: '2015-12-03 14:00:00'},
+    ],
+    4: [
+      {startsAt: '2015-12-04 12:00:00'},
+    ],
+    5: [
+      {startsAt: '2015-12-05 12:00:00'},
+    ],
+  }
+
+  t.deepEqual(tt.groupEventsByDays(events), expected, 'groups events by weekdaynum')
+
+  t.end()
+})
+
+test('calculateOverlap()', t => {
+  const expected = [
+    // startsAt, endsAt, _firstOverlapping, _overlaps
+    ['2015-12-03 12:00:00', '2015-12-03 13:00:00', false, 0],
+    ['2015-12-03 13:00:00', '2015-12-03 14:00:00', false, 0],
+    ['2015-12-03 15:00:00', '2015-12-03 17:00:00', false, 0],
+    ['2015-12-03 20:00:00', '2015-12-03 20:30:00', false, 0],
+    ['2015-12-04 12:00:00', '2015-12-04 13:00:00', true, 1],
+    ['2015-12-04 12:30:00', '2015-12-04 14:00:00', false, 1],
+    ['2015-12-04 15:00:00', '2015-12-04 17:00:00', true, 1],
+    ['2015-12-04 16:00:00', '2015-12-04 17:00:00', false, 1],
+    ['2015-12-05 12:00:00', '2015-12-05 13:00:00', true, 2],
+    ['2015-12-05 12:30:00', '2015-12-05 13:00:00', false, 2],
+    ['2015-12-05 12:45:00', '2015-12-05 15:00:00', false, 2],
+    ['2015-12-05 15:00:00', '2015-12-05 20:30:00', false, 0],
+  ]
+
+  const events = R.map(([startsAt, endsAt]) => {
+    return {
+      startsAt,
+      endsAt,
+    }
```

Comment:
Since this is a single statement, use `()` instead of `{)` and remove `return` !fix